### DISCUSSION TITLE
Error in calculating NRI Battery if on negative export price

### DISCUSSION THREAD

**Discussion by Rav3nholm** [JensenNick/huawei_solar_pees#43] [Forum]
I have used this pack since february and it has worked flawlessly!
Until February 7...

Between 12:00 and 16:00 we had negative price here in Sweden (SE3) and all hell broke loose on sensor.pv_economy_nri_battery.
From 142,43SEK down to -20131,93SEK in those 4 hours, see graph:

![PEES](https://github.com/JensenNick/huawei_solar_pees/assets/98101559/d99d29dd-e85b-4468-b757-c3888ce06e5d)

I think it has something to do with multiplying a negative value (as the export in regards to "cost" of charging the battery but would like some help...
I have not modified anything in this pack other than the input from electrical prices.

**Comment by JensenNick** [comment]
First, which version are you on? One of the reasons for introducing the suffix _ps, _ts and _tt was because HA was renaming sensors with the suffix _2 depending on the order the sensors were loaded. This could cause major issues like this. The new naming was introduced in version v1.0.0. You can se the version in the top of the yaml-fil (unless it is before I started adding the version no.)
![2024-04-09_08-48-25](https://github.com/JensenNick/huawei_solar_pees/assets/149193835/e3142419-df27-4dfe-b519-c5bdc63198bc)

You should have a look at the input sensors for `sensor.pv_economy_nri_battery` to determine which of the input sensors is causing the issue. As of the latest version the yaml code looks like this which is then feed into a utility meter with the same name, just without the suffix _ts.
![2024-04-09_08-49-18](https://github.com/JensenNick/huawei_solar_pees/assets/149193835/715ae04f-9b8f-4221-a29a-03f7bba14140)

**Comment by Rav3nholm** [comment]
Okey the version am not sure about beacuse I did not use your entire file as "package" but rather copied its different parts into my configuration. I already hade packages so I wrongly asumed you could not have multiple packages so yeah, that something I will fix later.
But started using PEES back in january/february I have the old naming.

`sensor.pv_economy_nri_battery` get a faulty value from `sensor.battery_charge_yield_sale_2` as seen here:
![PEES2](https://github.com/JensenNick/huawei_solar_pees/assets/98101559/b3344764-b5fe-4092-b58d-9f293fd13db0)

That sensor is in turn getting a faulty value from `sensor.battery_charge_yield_sale` as seen here:
![PEES3](https://github.com/JensenNick/huawei_solar_pees/assets/98101559/69355e79-5b50-4fd1-aca4-d25c38bbcbdc)

That sensors code is this:
![PEES5](https://github.com/JensenNick/huawei_solar_pees/assets/98101559/18b5a089-8888-45a1-b257-85ea4029183e)

As you see I have my own price sensor configured as `sensor.elec_price_on_production_export`
And just for good measure here is what the output of that sensor looks like on that day:
![PEES4](https://github.com/JensenNick/huawei_solar_pees/assets/98101559/bf17890f-6c6e-490f-870b-5bdc3f4c2db8)
